tools/check_format.py

```diff
@@ -153,6 +155,25 @@
 # yapf: enable
 
 
+# Map a line transformation function across each line of a file.
+# .bak temporaries.
+def replaceLines(path, line_xform):
+  # We used to use fileinput in the older Python 2.7 script, but this doesn't do
+  # inplace mode and UTF-8 in Python 3, so doing it the manual way.
+  output_lines = [line_xform(line) for line in readLines(path)]
+  pathlib.Path(path).write_text('\n'.join(output_lines), encoding='utf-8')
```

Comment:
i'll note that this needs to buffer the entire output set before writing, which seems wasteful of memory. But I guess this doesn't matter for this application?

Reply:
Yeah I don't think it matters, not likely to have MB level size of a source code file.

Reply:
Yep, that was my thinking as well.